isabelle/generated/SemanticsGuarantees.hs

```diff
@@ -46,18 +41,17 @@ instance Orderings.Order SemanticsTypes.Party where {
 instance Orderings.Linorder SemanticsTypes.Party where {
 };
 
-less_eq_Tok :: SemanticsTypes.Token -> SemanticsTypes.Token -> Bool;
-less_eq_Tok (SemanticsTypes.Token a b) (SemanticsTypes.Token c d) =
-  (if a < c then True else (if c < a then False else b <= d));
-
 less_eq_Token :: SemanticsTypes.Token -> SemanticsTypes.Token -> Bool;
-less_eq_Token a b = less_eq_Tok a b;
-
-less_Tok :: SemanticsTypes.Token -> SemanticsTypes.Token -> Bool;
-less_Tok a b = not (less_eq_Tok b a);
+less_eq_Token a b =
+  (case (a, b) of {
+    (SemanticsTypes.Token currencyA tokenA,
+      SemanticsTypes.Token currencyB tokenB)
+      -> (if currencyA < currencyB then True
+           else (if currencyB < currencyA then False else tokenA <= tokenB));
+  });
```

Comment:
Same here, we are not gaining anything by having a `case`, plus a `let` would be more appropriate. The functionality of pattern matching in the top level is there for a reason, why act like it doesn't exist?

Reply:
done